arch/common/sw_isr_common.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sw_isr_table.h>
+#include <arch/cpu.h>
+
+/*
+ * Common code for arches that use software ISR tables (CONFIG_GEN_ISR_TABLES)
+ */
+
+#ifdef CONFIG_DYNAMIC_INTERRUPTS
```

Comment:
Pedantry: you don't need to wrap the definition of non-statics with an #ifdef.  They will only enter the link if called, and the declaration is already wrapped.